p4-16/psa/psa.p4

```diff
@@ -568,12 +579,13 @@ control ComputeChecksum<H, M>(inout H hdr, inout M user_meta);
 
 control Deparser<H>(packet_out buffer, in H hdr);
 
-package PSA_Switch<H, M>(Parser<H, M> p,
-                         VerifyChecksum<H, M> vr,
+package PSA_Switch<H, M>(IngressParser<H, M> ip,
                          Ingress<H, M> ig,
+                         Deparser<H> igdep,
+                         EgressParser<H, M> ep,
                          Egress<H, M> eg,
                          ComputeChecksum<H, M> ck,
-                         Deparser<H> dep);
+                         Deparser<H> egdep);
```

Comment:
If checksum calculation and fixing up were allowed in ingress and egress, I don't see a reason why the compiler would need to impose any restrictions on where it was done.  If the programmer wanted it to be after every other modification, they could simply make it occur last sequentially in their ingress/egress apply block.

That said, I have no problems with allowing checksum fixup calls in the deparsers.  That will require making the headers 'inout' direction, else the header fields cannot be modified, but that is easy to do.